tests/test_TFUtil.py

```diff
@@ -469,6 +469,15 @@ def test_Data_copy_compatible_to_batch_feature_is_dynamic():
   assert_equal(t.get_size_dim_tag(0), dec.get_time_dim_tag())
 
 
+def test_Data_get_common_data():
+  d1 = Data(name='t', shape=(None, 32, 128), dtype='float32', batch_dim_axis=0, time_dim_axis=1, feature_dim_axis=3,
+            auto_create_placeholders=True)
+  d2 = Data(name='r', shape=(None, 32, 128), dtype='float32', batch_dim_axis=0, time_dim_axis=1, feature_dim_axis=3,
+            auto_create_placeholders=True)
+  common = Data.get_common_data([d1, d2], warnings_out=sys.stdout)
+  assert d1.shape == common.shape
```

Comment:
The test is actually wrong. Because you create individual placeholders for each, it will create two different independent sizes. So the common shape should be `(None, None, 32, 128)`.

Reply:
But that's the case for the combine layer, where I have two placeholders as an input.

Reply:
If you want that you get your expected result, you would need to add this:
```
d2.get_size_dim_tag(0).declare_same_as(d1.get_size_dim_tag(0))
```

Reply:
I'm speaking about the size placeholders, and their dimension tags. They should be the same. Or if they are not, the question is, why are they not the same. Anyway, `Data.get_common_shape` behaves correct here.

Reply:
But I need that behavior here
 https://github.com/rwth-i6/returnn/blob/da95d40a5c1affd54bc1c57d9820786f1fd12d8e/TFNetworkLayer.py#L5608

Reply:
As I explained, that code is correct there. See my comments on `get_common_data`. It's correct. (Or where do you think it is wrong? Please read my comments.)